scripts/python/grsolver/grsolver.py

```diff
@@ -0,0 +1,319 @@
+                eps=r[1]-r[0]
+                A0=1./a[0]**2/eps**2+da[0]/(2*a[0]**3.*eps)-1./(a[0]**2.*r[0]*eps)
+                B0=-(2./(a[0]**2.*eps**2)+4*np.pi*G/c**2.*(S_adm[0]/c**2.+rho_adm[0])+3./2.*K[0]**2.)
+                C0=1./(a[0]**2.*eps**2.)-da[0]/(2.*a[0]**3.*eps)+1./(a[0]**2.*r[0]*eps)
+                ##V[0]=A0
+                M[0,0]=A0+B0
+                M[0,1]=C0
+                AL=1/a[len(r)-1]**2/eps**2+da[len(r)-1]/(2*a[len(r)-1]**3.*eps)-1./(a[len(r)-1]**2.*r[len(r)-1]*eps)
+                BL=-(2./(a[len(r)-1]**2.*eps**2)+4*np.pi*G/c**2.*(S_adm[len(r)-1]/c**2+rho_adm[len(r)-1])+3./2.*K[len(r)-1]**2.)
+                CL=1./(a[len(r)-1]**2.*eps**2.)-da[len(r)-1]/(2.*a[len(r)-1]**3.*eps)+1./(a[len(r)-1]**2.*r[len(r)-1]*eps)
+                V[len(r)-1]=eps/r[len(r)-1]*CL
+                M[len(r)-1,len(r)-2]=AL
+                M[len(r)-1,len(r)-1]=BL+(1.-eps/r[len(r)-1])*CL
+
+                
+                ## THE REST OF THE MATRIX       
+                for i in range(1,len(r)-1):
+                        A=1/a[i]**2/eps**2+da[i]/(2*a[i]**3.*eps)-1./(a[i]**2.*r[i]*eps)
+                        B=-(2./(a[i]**2.*eps**2)+4*np.pi*G/c**2.*(S_adm[i]/c**2.+rho_adm[i])+3./2.*K[i]**2.)
+                        C=1./(a[i]**2.*eps**2.)-da[i]/(2.*a[i]**3.*eps)+1./(a[i]**2.*r[i]*eps)
+                        M[i,i-1]=A
+                        M[i,i]=B
+                        M[i,i+1]=C
+                def f(x):
+                        return np.dot(M,x)+V
+                sol = optimize.root(f,100*np.ones(len(r)))
+                alpha=sol.x
+        
+        
+################# Solve for shift
+                beta=-a**2./2.*alpha*r*K                                                
+                return a,K,alpha,beta
+                
+        def CalculateADM(self,r,a,K,alpha,beta):        
+                rho_adm=np.zeros(len(r))
+                P_adm=np.zeros(len(r))
+                S_adm=np.zeros(len(r))
+                Srr_adm=np.zeros(len(r))
+
+                for i in range(len(r)):
+                        #upper metric
+                        g00=-1./alpha[i]**2.   
+                        g0r=beta[i]/alpha[i]**2.                
+                        gamma2=1./(1-self.v_int(r[i])**2./c**2.)
+                        rho_adm[i]=alpha[i]**2.*((self.rho_int(r[i])+self.p_int(r[i])/c**2.)*gamma2-self.p_int(r[i])/c**2.*g00)
+                        P_adm[i]=alpha[i]*beta[i]*((self.rho_int(r[i])+self.p_int(r[i])/c**2.)*gamma2+self.p_int(r[i])/c**2.*g00)+alpha[i]*((self.rho_int(r[i])+self.p_int(r[i])/c**2.)*gamma2*self.v_int(r[i])+self.p_int(r[i])/c**2.*g0r)   ## r-component  (upper index)
+                        S_adm[i]=(alpha[i]**2.*gamma2-1.)*self.rho_int(r[i])*c**2.+(alpha[i]**2.*gamma2+2.)*self.p_int(r[i])
+                        Srr_adm[i]=self.rho_int(r[i])*self.v_int(r[i])**2.*gamma2*a[i]**4.+self.p_int(r[i])*a[i]**4.*(1./a[i]**2-beta[i]**2./alpha[i]**2.+self.v_int(r[i])**2./c**2.*gamma2)
```

Comment:
This code is getting a bit difficult to read.. it might be good practice for us to adopt a code formatting strategy for python like we do with `clang-format`. I think `autopep8` is popular, even if I am not the hugest pep8 fan.

Reply:
I added an issue #163 . That should be a separate MR at a later date.